src/main/java/software/amazon/smithy/lsp/ext/FileCachingCollector.java

```diff
@@ -63,7 +63,12 @@ public Map<ShapeId, Location> collectDefinitionLocations(Model model) {
         this.membersToUpdateMap = new HashMap<>();
 
         for (ModelFile modelFile : this.fileCache.values()) {
-            collectContainerShapeLocationsInModelFile(modelFile);
+            try {
+                collectContainerShapeLocationsInModelFile(modelFile);
+            } catch (Exception e) {
+                throw new RuntimeException("Exception while collecting container shape locations in model file: "
```

Comment:
**thought:** Just wrapping the underlying one to add the extra information about the problematic file. This might make it easier to pinpoint the piece of syntax that the file collector is struggling with.